appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1662,9 +1662,14 @@ private void deployAll() throws GlassFishException {
                         // Convert the URL to a URI for use with the deploy method
                         URI artefactURI = deploymentMapEntry.getValue().toURI();
 
-                        deployer.deploy(artefactURI, "--availabilityenabled",
-                                "true", "--contextroot",
-                                deploymentMapEntry.getKey(), "--force=true", "--loadOnly", "true");
+                        String artefactName = getFileName(artefactURI.getPath());
+                        String name = artefactName.substring(0, artefactName.length() - 4);
```

Comment:
Yes, this will break in so many ways, I cannot count 👍

Reply:
Thanks I understand. May be try to fix this by writing artifact with actual file name to a temporary directory instead of temp file?